### PR TITLE
Refactor to standardized parsing format

### LINKED ISSUE #200: Fix invocation error bug for empty commands
Solution: Refactor all commands to be one word. E.g. "client add" -> "cltadd", "country add" -> "ctyadd"

Add test cases for all commands where the rest of the command is empty: "client add", etc.

### PR DESCRIPTION
**Pull request by tankangliang**
## Description

Changed previous implementation of parsing for `client note` commands to use the same regex.

Fixes #200 

## Testing

Commands starting with `client COMMAND` should not throw `InvocationError` anymore.

### CODE REVIEW

File: src/test/java/seedu/address/logic/parser/AddressBookParserTest.java [hunk 1]
```diff
@@ -102,10 +190,9 @@ public void parseCommand_unknownCommand_throwsParseException() {
...
-    public void parseCommand_view() throws Exception {
-        ClientViewCommand command = (ClientViewCommand) parser.parseCommand(
-                ClientViewCommand.COMMAND_WORD + " " + INDEX_FIRST_CLIENT.getOneBased());
-        assertEquals(new ClientViewCommand(INDEX_FIRST_CLIENT), command);
-    }
+
+
+
+
+
```
Comment (qwoprocks) [inline src/test/java/seedu/address/logic/parser/AddressBookParserTest.java]: Too many new lines here

File: src/test/java/seedu/address/logic/parser/AddressBookParserTest.java [hunk 2]
```diff
@@ -66,19 +64,109 @@ public void parseCommand_edit() throws Exception {
...
+    public void parseClientCommands_viewClient() throws Exception {
+        ClientViewCommand command = (ClientViewCommand) parser.parseCommand(
+                ClientViewCommand.COMMAND_WORD + " " + INDEX_FIRST_CLIENT.getOneBased());
+        assertEquals(new ClientViewCommand(INDEX_FIRST_CLIENT), command);
+    }
+
+    @Test
+    public void parseClientCommands_unrecognisedInput_throwsParseException() {
+        assertThrows(ParseException.class, String.format(MESSAGE_INVALID_COMMAND_FORMAT,
+                HelpCommand.MESSAGE_USAGE), () -> parser.parseCommand("client "));
```
Comment (qwoprocks) [inline src/test/java/seedu/address/logic/parser/AddressBookParserTest.java:93]: Can add one more assert for just "client" without the space? Just in case.

Same for client note test below

File: src/test/java/seedu/address/logic/parser/AddressBookParserTest.java [hunk 3]
```diff
@@ -66,19 +64,109 @@ public void parseCommand_edit() throws Exception {
...
+                HelpCommand.MESSAGE_USAGE), () -> parser.parseCommand("client note "));
+    }
+
+    @Test
+    public void parseClientNoteCommands_unknownCommand_throwsParseException() {
+        assertThrows(ParseException.class, MESSAGE_UNKNOWN_COMMAND, () ->
+                parser.parseCommand("client note unknownCommand"));
+    }
+
+
```
Comment (qwoprocks) [inline src/test/java/seedu/address/logic/parser/AddressBookParserTest.java]: extra new line here

File: src/main/java/seedu/address/logic/parser/AddressBookParser.java [hunk 4]
```diff
@@ -128,46 +127,63 @@ private Command parseCountryCommands(String input) throws ParseException {
...
-        if (nextWord.equals("add")) {
-            // gather rest of the arguments:
-            StringBuilder stringBuilder = new StringBuilder();
-            while (stringTokenizer.hasMoreTokens()) {
-                stringBuilder.append(stringTokenizer.nextToken()).append(" ");
-            }
-
-            commandWord += (" " + nextWord);
-            arguments = " " + stringBuilder.toString();
+        final String commandType = matcher.group("commandType");
```
Comment (qwoprocks) [inline src/main/java/seedu/address/logic/parser/AddressBookParser.java:135]: Can rename this to clientCommandType? and the country one to countryCommandType?

File: src/main/java/seedu/address/logic/parser/AddressBookParser.java [hunk 5]
```diff
@@ -128,46 +127,63 @@ private Command parseCountryCommands(String input) throws ParseException {
...
+     * @throws ParseException if input does not conform to expected format
+     */
+    private Command parseClientNoteCommands(String input) throws ParseException {
+        final Matcher matcher = BASIC_COMMAND_FORMAT.matcher(input.trim());
+        if (!matcher.matches()) {
+            throw new ParseException(
+                    String.format(MESSAGE_INVALID_COMMAND_FORMAT, HelpCommand.MESSAGE_USAGE));
+        }
+
+        final String commandType = matcher.group("commandType");
```
Comment (qwoprocks) [inline src/main/java/seedu/address/logic/parser/AddressBookParser.java:179]: rename this to clientNoteCommandType?

**Review by qwoprocks — CHANGES_REQUESTED** [review]

**Review by qwoprocks — APPROVED** [review]
LGTM
